### PR TITLE
new (fixed) module: entrezdirect/xtract

### LINKED ISSUE #1832: new module: entrezdirect/xtract
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by alxndrdiaz**
## PR checklist

Closes #1832 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by alxndrdiaz** [comment]
@muffato @jfy133 Had to remove the previous pull request because I broke it by merging other branches (same for esearch).

File: modules/entrezdirect/xtract/main.nf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    'quay.io/biocontainers/entrez-direct:16.2--he881be0_1' }"
+
+    input:
+    tuple val(meta), path(xml_input)
+    val pattern
+    val element
+    val sep
+
+    output:
+    tuple val(meta), path("*.txt"), emit: xtract_table
```
Comment (jfy133) [inline modules/entrezdirect/xtract/main.nf:17]:
```suggestion
    tuple val(meta), path("*.txt"), emit: txt
```

File: tests/modules/entrezdirect/xtract/test.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+- name: entrezdirect xtract test_entrezdirect_xtract_assembly
+  command: nextflow run ./tests/modules/entrezdirect/xtract -entry test_entrezdirect_xtract_assembly -c ./tests/config/nextflow.config  -c ./tests/modules/entrezdirect/xtract/nextflow.config
+  tags:
+    - entrezdirect/xtract
+    - entrezdirect
+  files:
+    - path: output/entrezdirect/test_assembly.txt
+      md5sum: 17073db7c14890aef30f5e3f2d003d32
+    - path: output/entrezdirect/test_assembly.xml
+      md5sum: ddbf36702303ef805a19f81754eb9c92
```
Comment (jfy133) [inline tests/modules/entrezdirect/xtract/test.yml:10]: Is this produced by xtract? If not I would remove it for both tests.

**Review by jfy133 — APPROVED** [review]
Couple of comments but looks good ot me

File: modules/entrezdirect/xtract/meta.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+name: "entrezdirect_xtract"
+description: Queries an NCBI database using an UID
+keywords:
+  - public datasets
```
Comment (jfy133) [inline modules/entrezdirect/xtract/meta.yml:4]:
```suggestion
  - public datasets
  - entrez
  - search
  - ncbi
```

File: tests/modules/entrezdirect/xtract/main.nf [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        []
+    ]
+    database = 'assembly'
+
+    pattern_in = 'DocumentSummary'
+    element_in = 'SpeciesName BioprojectAccn FtpPath_GenBank'
+    delim = ","
+
+    ENTREZDIRECT_ESUMMARY ( input, database )
+    ENTREZDIRECT_XTRACT ( ENTREZDIRECT_ESUMMARY.out.xml_esummary, pattern_in, element_in, delim )
```
Comment (muffato) [inline tests/modules/entrezdirect/xtract/main.nf:25]: @alxndrdiaz . I know it was accepted as `xml_esummary`, but I think the output in `entrezdirect/esummary` should be simply called `xml` in the module and here.

Comment (alxndrdiaz) [reply tests/modules/entrezdirect/xtract/main.nf:25]: Done , here is the PR: [renamed output in entrezdirect/esearch: xml #1928](https://github.com/nf-core/modules/pull/1928).

Comment (muffato) [inline tests/modules/entrezdirect/xtract/main.nf:25]:
```suggestion
    ENTREZDIRECT_XTRACT ( ENTREZDIRECT_ESUMMARY.out.xml, pattern_in, element_in, delim )
```

File: tests/modules/entrezdirect/xtract/main.nf [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        []
+    ]
+    database = 'genome'
+
+    pattern_in = 'DocumentSummary'
+    element_in = 'TaxId Organism_Name Project_Accession Assembly_Accession'
+    delim = ","
+
+    ENTREZDIRECT_ESUMMARY ( input, database )
+    ENTREZDIRECT_XTRACT ( ENTREZDIRECT_ESUMMARY.out.xml_esummary, pattern_in, element_in, delim )
```
Comment (muffato) [inline tests/modules/entrezdirect/xtract/main.nf:45]:
```suggestion
    ENTREZDIRECT_XTRACT ( ENTREZDIRECT_ESUMMARY.out.xml, pattern_in, element_in, delim )
```

**Review by muffato — COMMENTED** [review]
One #1928 is approved

File: tests/modules/entrezdirect/xtract/test.yml (hunk 2 above)
Comment (alxndrdiaz) [reply tests/modules/entrezdirect/xtract/test.yml:10]: No, it is only the input it takes, so it could be removed.

**Comment by alxndrdiaz** [comment]
> One #1928 is approved

Done.

**Comment by alxndrdiaz** [comment]
This module is part of the Google Summer of Code 2022 project: [Conversion of the BlobToolKit pipeline to Nextflow](https://summerofcode.withgoogle.com/myprojects/details/v89lHomV), GitHub repository: https://github.com/sanger-tol/blobtoolkit

**Pull request merged**